packages/hardhat-core/test/internal/hardhat-network/provider/modules/eth/methods/getStorageAt.ts

```diff
@@ -229,14 +233,73 @@ describe("Eth module", function () {
                 assert.strictEqual(
                   await this.provider.send("eth_getStorageAt", [
                     exampleContract,
-                    numberToRpcQuantity(0),
+                    numberToRpcStorageSlot(0),
                   ]),
                   "0x000000000000000000000000000000000000000000000000000000000000007c"
                 );
               });
             });
           });
         });
+
+        describe("validation", function () {
+          it("should accept valid storage slot arguments", async function () {
+            assert.strictEqual(
+              await this.provider.send("eth_getStorageAt", [
+                "0x0101010101010101010101010101010101010101",
+                "0x0000000000000000000000000000000000000000000000000000000000000000",
+              ]),
+              "0x0000000000000000000000000000000000000000000000000000000000000000"
+            );
+
+            assert.strictEqual(
+              await this.provider.send("eth_getStorageAt", [
+                "0x0101010101010101010101010101010101010101",
+                "0xcd39aa866fd639607c7241f617cf83f33c646551f3d205f2905c5abacca2db85",
```

Comment:
Do these bytestrings have any specific relevance? How does the expected output relate to these inputs?

Reply:
Just a random storage slot, and an address that is (almost surely) empty. Added some comments explaining this.